### PR TITLE
Read layer metadata from raster GDB layers

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: src/core/providers/gdal/qgsgdalprovider.cpp [hunk 1]
```diff
@@ -485,6 +486,29 @@ void QgsGdalProvider::reloadProviderData()
   ( void )initIfNeeded();
 }
 
+void QgsGdalProvider::loadMetadata()
+{
+  // Set default, may be overridden by stored metadata
+  mLayerMetadata.setCrs( crs() );
+
+  if ( mDriverName == QLatin1String( "OpenFileGDB" ) )
```
Comment (nirvn) [inline src/core/providers/gdal/qgsgdalprovider.cpp:494]: If someone wanted to expand this beyond OpenFileGDB, are there other drivers worth exploring? This is a general question, not bound to this PR :)

**Review by nirvn — APPROVED** [review]
Nice one. Is there a minimum gdal version involved here?

**Comment by nyalldawson** [comment]
> Is there a minimum gdal version involved here?

It's kind of implicit -- only 3.7 + will allow opening layers using the OpenFileGDB driver, so this code path can only be hit on those versions

File: src/core/providers/gdal/qgsgdalprovider.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/providers/gdal/qgsgdalprovider.cpp:494]: > If someone wanted to expand this beyond OpenFileGDB, are there other drivers worth exploring? This is a general question, not bound to this PR :)

Yes, if there's other GDAL formats which have metadata available to them which we can map across to QGIS metadata fields!

**Comment by nirvn** [comment]
@nyalldawson , maybe it's worth adding at least a note in the  loadMetadata() function to let people reading the code know it is gdal >= 3.7 compatible. That'll be friendlier than having to read the whole code path to it :)

**Pull request merged**